tools/rw-benchmark/linkedmap.go

```diff
@@ -0,0 +1,310 @@
+// Copyright 2022 The etcd Authors
```

Comment:
Is there externally maintained representation that could be a replacement...

It's significant piece of generic code that would be good if we delegated elsewhere... 
Something from https://github.com/zyedidia/generic ?

If not - we should put it into ./pkg/linkedmap ...

Reply:
I'm only briefly taking a look between meetings right now, but it seems to me that the LinkedMap is only an intermediary format between CSVs and float slices (`convertBenchmarkData`). Can't this be directly read into the slice format?

Reply:
> Is there externally maintained representation that could be a replacement...
> 
> It's significant piece of generic code that would be good if we delegated elsewhere... Something from https://github.com/zyedidia/generic ?

Ah:), it's coming from my personal repo [linkedmap](https://github.com/ahrtr/gocontainer/tree/master/map/linkedmap). I also added [a comment](https://github.com/ahrtr/etcd/blob/c41fa9a04f4178d3da81a595ddc7208de7c60e40/tools/rw-benchmark/linkedmap.go#L38-L39) in this PR. In the beginning, I did introduce a dependency on [ahrtr/gocontainer/linkedmap]([linkedmap](https://github.com/ahrtr/gocontainer/tree/master/map/linkedmap); but on second thought, I don't think it's good to depend a personal repo, so I copied the file into etcd directly. 

If there is no objection, I can remove linkedmap.go (and linkedmap_test.go) and introduce dependency on [ahrtr/gocontainer/linkedmap]([linkedmap](https://github.com/ahrtr/gocontainer/tree/master/map/linkedmap). 



> Can't this be directly read into the slice format?

I intentionally did it in order to decouple the data parser and data render. It's possible that we may change the CSV data format, and in that case we do not need to change the data render part.

Reply:
Anyway, I just removed `linkedmap.go` and `linkedmap_test.go`, and added dependency on [gocontainer#linkedMap](https://github.com/ahrtr/gocontainer#linkedMap). 

Note it will NOT populate the go.mod/go.sum files of any etcd users (e.g. clientv3 users, or embedded etcdserver users).

Reply:
Please only pay attention to the latest two new added commits if you have already reviewed this PR. thx

Reply:
added two more minor commits today.